lib/template-mixins.js

```diff
@@ -7,19 +7,35 @@ var Hogan = require('hogan.js'),
     _ = require('underscore'),
     moment = require('moment');
 
+var deprecate = require('deprecate');
+
 // This returns a middleware that places mixins against the `res.locals` object.
 //
-// It should be given:
-// - fields: the data needed to generate mixins options, etc.
 // - options:
-//   - viewDirectory: the folder in which templates are found in.
+//   - viewsDirectory: the folder in which templates are found in.
 //   - viewEngine: the type of view, defaults to 'html'.
 //   - sharedTranslationsKey: used to find translations relatively within
 //     the translations.json. Useful for field and button labels.
-module.exports = function (fields, options) {
-
-    fields = fields || {};
+//   - translate: a translate function
+//   - fields: static field configuration (overridden by `res.locals.options.fields`)
+module.exports = function (options, deprecated) {
+    var optionsKeys = ['viewsDirectory', 'viewEngine', 'sharedTranslationsKey', 'translate'];
+
+    var deprecateStr = 'hmpo-template-mixins: Passing field config to mixins is deprecated';
+    deprecateStr += '\nDynamic field config will be loaded from res.locals.options.fields';
+    deprecateStr += '\nStatic field config can be passed in as `options.fields`';
+
+    if (arguments.length === 1 && !_.intersection(Object.keys(options), optionsKeys)) {
```

Comment:
`_.intersection()` will return an empty array if the two sets have no overlaps. Use `_.isEmpty()` or `.length == 0`